sklearn/feature_selection/tests/test_rfe.py

```diff
@@ -109,6 +109,34 @@ def test_rfe():
     assert_array_almost_equal(X_r, X_r_sparse.toarray())
 
 
+def test_rfe_negative_n_features():
+    clf = SVC(kernel="linear")
+    with pytest.raises(ValueError, match=r"n_features_to_select must be *"):
+        RFE(estimator=clf, n_features_to_select=-1, step=0.1)
+
+
+def test_rfe_percent_n_features():
+    # test that the results are the same
+    generator = check_random_state(0)
+    iris = load_iris()
+    X = np.c_[iris.data, generator.normal(size=(len(iris.data), 6))]
+    y = iris.target
+    # there are 10 features in the data. We select half.
+    clf = SVC(kernel="linear")
+    rfe_num = RFE(estimator=clf, n_features_to_select=5, step=0.1)
+    rfe_num.fit(X, y)
+
+    rfe_perc = RFE(estimator=clf, n_features_to_select=0.5, step=0.1)
+    rfe_perc.fit(X, y)
+
+    rfe_other = RFE(estimator=clf, n_features_to_select=2, step=0.1)
```

Comment:
This case is redundant with the one above I think?

Could you also check with a small positive float (e.g. `n_features_to_select=0.01`) and assert the number of output features.